src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -404,6 +410,21 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
             }
         }
 
+        private bool CheckPlatformVersions()
+        {
+            IEnumerable<NuGetFramework> badPlatforms = _request.Project.TargetFrameworks.Select(tfm => tfm.FrameworkName).Where(fw => !string.IsNullOrEmpty(fw.Platform) && (fw.PlatformVersion == FrameworkConstants.EmptyVersion));
+            if (badPlatforms.Any())
+            {
+                NuGetFramework fw = badPlatforms.First();
```

Comment:
Why don't log all badPlatforms once, not just first one?

Reply:
I didn't wanna be spammy. Anyone else have thoughts on this?

Reply:
I like both points being made. Hmm, how to balance?

- Be more spammy with Verbosity Detailed
- Show the count somehow like "<First one> ...& 3 more"
- Show a generic message like "blah blah... the following platforms: " so it looks more like a stack trace (which devs are used to)

Reply:
Given that this is part of Restore. Would showing all bad platforms really help package consumers? They won't be able to restore the package either way, and, unless they have control over the dependency, they are pretty much blocked. 

---
In reply to: [480475287](https://github.com/NuGet/NuGet.Client/pull/3616#discussion_r480475287) [](ancestors = 480475287)

Reply:
Large framework count is *extremely* uncommon. 
I think we can just log them all. 

I don't think we should be moving the goal post if we already have the info that we need.